scipy/integrate/_quadrature.py

```diff
@@ -1035,3 +1038,131 @@ def newton_cotes(rn, equal=0):
     fac = power*math.log(N) - gammaln(p1)
     fac = math.exp(fac)
     return ai, BN*fac
+
+
+def _qmc_quad_iv(func, ranges, n_points, qmc_engine, args):
+
+    # lazy import to avoid issues with partially-initialized submodule
+    if not hasattr(qmc_quad, 'qmc'):
+        from scipy import stats
+        qmc_quad.stats = stats
+    else:
+        stats = qmc_quad.stats
+
+    if not callable(func):
+        message = "`func` must be callable."
+        raise TypeError(message)
+
+    ranges = np.atleast_2d(ranges)
+    dim = ranges.shape[0]
+
+    if args is None:
+        args = tuple()
+
+    try:
+        len(args)
+    except TypeError:
+        args = (args,)
+
+    try:
+        func(*ranges[:, 0], *args)
+    except Exception as e:
+        message = ("`func` must evaluate the integrand at points within and "
+                   "including the boundaries of the integration range; "
+                   "e.g. `func(*ranges[:, 0])` and `func(*ranges[:, 1])` "
+                   "must return the integrand at the lower and upper "
+                   "integration limits.")
+        raise ValueError(message) from e
+
+    try:
+        func(*ranges, *args)
+        vfunc = func
+    except Exception as e:
+        wmessage = ("Exception encountered when attempting vectorized call to "
+                    f"`func`: {e}. `func` should accept arrays `x0, ..., xn` "
+                    "for better performance.")
+        emessage = ("Exception encountered when attempting vectorized call to "
+                    f"`func`: {e}, and `args` is not `None`. `func` must "
+                    "accept arrays `x0, ..., xn` if `args` is not `None`.")
+        if args != tuple():
+            raise ValueError(emessage)
+        else:
+            warnings.warn(wmessage)
+        vfunc = np.vectorize(func)
+
+    n_points_int = np.int64(n_points)
+    if n_points != n_points_int:
+        message = "`n_points` must be an integer."
+        raise TypeError(message)
+
+    if qmc_engine is None:
+        qmc_engine = stats.qmc.Halton(dim)
+    elif not isinstance(qmc_engine, stats.qmc.QMCEngine):
+        message = "`qrng` must be an instance of scipy.stats.qmc.QMCEngine."
+        raise TypeError(message)
+
+    return vfunc, ranges, n_points_int, qmc_engine, args, stats
+
+
+QMCQuadResult = namedtuple('QMCQuadResult', ['integral', 'standard_error'])
+
+
+def qmc_quad(func, ranges, *, n_points=1024, qrng=None, args=None):
+    """
+    Compute an integral in N-dimensions using Quasi-Monte Carlo quadrature.
+
+    Parameters
+    ----------
+    func : callable
+        The function to be integrated. Must accept separate arguments
+        ``x0, ..., xn`` corresponding with the coordinate in each dimension.
+        May also accept additional positional arguments specified with
+        `args`.
```

Comment:
It would be more natural for `func` to accept a single array, but I was trying to be consistent with `nquad`.

Reply:
I think we should break from the interface of the `integrate` functions here. Since QMC integration is often used in higher dimensions, it just doesn't make sense to accept all the variables as separate inputs. Also, it would probably be better to accept the lower bounds of all variables in one input, and the upper bounds in another like `qmc.scale`, `multivariate_normal.cdf`, `optimize.Bounds`, etc... This would eliminate a lot of transposes in the code and hopefully better match what users would want in higher dimensions.

Reply:
Done.